src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -45,55 +55,135 @@ public Command setState(RobotState state) {
 	//@formatter:off
 	public Command idle() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE)
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+				funnelStateHandler.setState(FunnelState.STOP)
+				//elevatorRoller.STOP
+				//intakeRoller.STOP
+				//flywheel.DEFAULT
+				//intakePivot.UP
+				//elevator.IDLE
 		);
 	}
 
 	public Command intake() {
 		return new ParallelCommandGroup(
+				new SequentialCommandGroup(
+						new ParallelCommandGroup(
+								//intakePivot.INTAKE
+								//intakeRoller.INTAKE
+								funnelStateHandler.setState(FunnelState.STOP)
+						).withTimeout(3), // .until(() -> isNoteInFunnel
+						new ParallelCommandGroup(
+								funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER)
+								//intakePivot.UP
+								//intakeRoller.NOTE_TO_SHOOTER
+						).until(this::isNoteInShooter),
+						new ParallelCommandGroup(
+								//intakeRoller.STOP
+								funnelStateHandler.setState(FunnelState.STOP)
+						)
+				),
 				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.NOTE))
+				//flywheel.DEFAULT
+				//elevator.IDLE
+				//elevatorRoller.IDLE
 		);
 	}
 
 	public Command preSpeaker() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER))
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER)),
+				//flywheel.ACCELERATE/SHOOT
+				funnelStateHandler.setState(FunnelState.STOP)
+				//intakeRoller.STOP
+				//elevator.IDLE
+				//intakePivot.UP
```

Comment:
```suggestion
				//intakePivot.IDLE
```
change in all places